__tests__/routes/note.test.js

```diff
@@ -0,0 +1,372 @@
+const request = require('supertest');
+const express = require('express');
+const db = require('../../data/db-config');
+const authRequired = require('../../api/middleware/authRequired');
+const handleError = require('../../api/middleware/handleError');
+const notesRouter = require('../../api/notes/noteRouter');
+
+beforeAll(async () => {
+  await db.migrate.rollback();
+  await db.migrate.latest();
+});
+beforeEach(async () => await db.seed.run());
+
+afterAll(async () => await db.destroy());
+
+afterEach(() => jest.clearAllMocks());
+
+jest.mock('../../api/middleware/authRequired', () =>
+  jest.fn((req, res, next) => next())
+);
+
+jest.mock('../../api/middleware/permissionsRequired', () => ({
+  adminRequired: jest.fn((req, res, next) => next()),
+  superAdminRequired: jest.fn((req, res, next) => next()),
+}));
+
+const app = express();
+app.use(express.json());
+app.use('/notes', notesRouter);
+app.use(handleError);
+describe('Sanity Checks', () => {
+  test('matchers are working', () => {
+    expect(true).toBe(true);
+    expect(20 - 5).toBe(15);
+    expect(9 + 10).not.toEqual(21);
+  });
+
+  test('test environment is being used', () => {
+    expect(process.env.NODE_ENV).toBe('test');
+  });
+});
+
+describe('Notes Router', () => {
+  describe('[GET] /notes', () => {
+    describe('succeed', () => {
+      let res;
+      beforeAll(async () => {
+        res = await request(app).get('/notes');
+      });
+
+      it('requires authentication', () => {
+        expect(authRequired).toBeCalled();
+      });
+
+      it('responds with status 200', async () => {
+        const expected = 200;
+        const actual = res.status;
+        expect(actual).toBe(expected);
+      });
+    });
+    describe('failed', () => {});
+  });
+
+  describe('[GET] /notes/note_id', () => {
+    describe('succeed', () => {
+      let res;
+      beforeAll(async () => {
+        res = await request(app).get('/notes/1');
+      });
+
+      it('requires authentication', () => {
+        expect(authRequired).toBeCalled();
+      });
+
+      it('responds with status 200', async () => {
+        const expected = 200;
+        const actual = res.status;
+        expect(actual).toBe(expected);
+      });
+
+      it('responds with non empty object', async () => {
```

Comment:
Will this test break when production goes live or new seeds are tested?

Reply:
The tests for /notes/mentee/:mentee_id endpoint will break if the seed file changes under these conditions:
_less than 5 note entries per mentee_id
_the mentee_id changes

As for the post and put endpoints, the tests should not break if the schema structure remains unchanged.

Finally, the get and get/:note_id, the tests should not break if the schema or seed changes.